cloudformation/templates/gridss_worker_instance/worker.yaml

```diff
@@ -0,0 +1,221 @@
+      SecurityGroupIds:
+        - !Ref instanceSecurityGroup
+      SubnetId: !Ref instanceSubnet
+      LaunchTemplate:
+        LaunchTemplateId: !Ref workerInstanceLauchTemplate
+        Version: !GetAtt 
+          - workerInstanceLauchTemplate
+          - LatestVersionNumber
+      KeyName: alexis-dev
+      Tags:
+        - Key: Source
+          Value: CloudFormation
+        - Key: Creator
+          Value: !Ref userName
+        - Key: Env
+          Value: Test
+        - Key: Name
+          Value: Worker instance
+        - Key: Stack
+          Value: !Ref AWS::StackId
+        - Key: Project
+          Value: gridss-purple-linx
+      UserData:
+        Fn::Base64: !Sub |
+            #!/bin/bash -xe
+            # Set time/logs to melbourne time
+            ln -sf /usr/share/zoneinfo/Australia/Melbourne /etc/localtime
+            # create mount point directory
+            mkdir /mnt/xvdh
+            # create ext4 filesystem on new volume           
+            mkfs -t ext4 /dev/xvdh
+            # add an entry to fstab to mount volume during boot
+            echo "/dev/xvdh       /mnt/xvdh   ext4    rw,suid,dev,exec,auto,user,async,nofail 0       2" >> /etc/fstab
+            # mount the volume on current boot
+            mount -a
+            # create a user folder for both users on this directory
+            mkdir /mnt/xvdh/ssm-user
+            mkdir /mnt/xvdh/ec2-user
+            # change the owner so the user (via SSM) has access
+            chown -R ssm-user /mnt/xvdh/ssm-user
+            chown -R ec2-user /mnt/xvdh/ec2-user
+            # install dev tools for later use
+            yum update -y -q
+            yum groups mark install "Development Tools"
+            yum update -y -q 
+            yum groupinstall -y 'Development Tools' 
+            # install docker
+            amazon-linux-extras install docker
+            # Move docker to larger diskspace
+            mv /var/lib/docker /mnt/xvdh/docker
+            # Link mount to docker container
+            ln -s /mnt/xvdh/docker /var/lib/docker
+            # Add users to docker
+            usermod -a -G docker ssm-user
+            usermod -a -G docker ec2-user
+            service docker start
+            # Install s3fuse from amazon
+            amazon-linux-extras install epel -y
+            yum install s3fs-fuse -y
+            # Sync gridss data from s3 bucket
+            aws s3 sync --quiet s3://umccr-misc-temp/gridss_hg19_refdata/hg19/ /mnt/xvdh/hg19_gridss_ref_data/
+            # Install container registry helper
+            yum install amazon-ecr-credential-helper -y
+            # Add configuration to docker config
+            su - "ec2-user" -c 'mkdir -p $HOME/.docker && echo "{ \"credsStore\" : \"ecr-login\" }" >> $HOME/.docker/config.json'
+            # Pull the gridss-purple-linx container image from the ECR
+            su - "ec2-user" -c 'docker pull 843407916570.dkr.ecr.ap-southeast-2.amazonaws.com/gridss-purple-linx:2.7.3'
```

Comment:
@alexiswl Could you parametrise this out somehow? I almost missed it and can be fiddly to update if buried down in userdata :-S

Reply:
As in the ECR container registry link and the docker image and tag names?

Reply:
Yeah, sth like this?: https://github.com/umccr/umccrise/blob/master/buildspec.yml#L4

Reply:
I've used the AWS::AccountId and AWS::Region envs, which I get might cause some problems if the container doesn't exist in that account / region but we can always just duplicate the container to multiple registries, it makes more sense to grab it from the registry from the corresponding account and fail if it doesn't contain the repo.